src/Campaigns/Models/Campaign.php

```diff
@@ -84,6 +95,18 @@ public function forms(): ModelQueryBuilder
             })->where('campaign_forms.campaign_id', $this->id);
     }
 
+    /**
+     * @unreleased
+     */
+    public function legacyForms(): ModelQueryBuilder
```

Comment:
Honestly, I'm not a big fan of this. The `forms` method should return all forms associated with the campaign regardless of whether it's V2 or V3. I understand that there are fundamental differences between the two, and I also know this would require extra effort to accommodate what I'm suggesting. @kjohnson any ideas?

Maybe the best approach to get what I want here is to refactor the  `defaultForm` method. Using the `give_campaign_forms` table we get the default form ID, then we check the type of the form (we can do all this in one query by joining `wp_posts`, `wp_postmeta` and `give_campaign_forms` and including `formBuilderSettings` meta key to the query - if the returned value is null it means that we have V2 form ), and then we return the correct form model.

Reply:
I just came across this while working on the Default Form select setting.

When I wrote the `forms()` method the assumption, if I remember correctly, was that these would be v3 forms. We'll need to update this to return both v2 and v3 forms.

Reply:
This would be difficult to do as a single query with support for hydrating models for multiple types. We could extend the `ModelQueryBuilder` class and do some switching logic in the `getRowAsModel()` method.

Reply:
@alaca I agree that this is not the best approach, but as we already have a [**task**](https://stellarwp.atlassian.net/browse/GIVE-1949) to update how the default form ID will be stored and retrieved from DB, I think we can leave it as it is right now to move forward and then revisit it later, even because a query written now well need get changed when we change the place to store the default form ID.

Reply:
Hmm okay, that makes sense. I'm working on that task so I will refactor this anyway.